File: VB.NET2010/Viewer/MergeSplitDeletePDF.vb
Public Class MergeSplitDeletePDF

    Public iType As Integer
    Public strMergeSource1 As String
    Public strMergeSource2 As String

    Public strSplitSource As String
    Public iSplitStartPage As Integer
    Public strDeletePageSource As String
    Public iDeletePageNo As Integer



    Private Sub button1_Click(sender As System.Object, e As System.EventArgs) Handles button1.Click
        OpenFileDialog1.Filter = "PDF File|*.PDF||"
        If OpenFileDialog1.ShowDialog() = System.Windows.Forms.DialogResult.OK Then

            txtmergesource1.Text = OpenFileDialog1.FileName
        End If

    End Sub

    Private Sub button2_Click(sender As System.Object, e As System.EventArgs) Handles button2.Click
        OpenFileDialog1.Filter = "PDF File|*.PDF||"
        If OpenFileDialog1.ShowDialog() = System.Windows.Forms.DialogResult.OK Then

            txtmergesource2.Text = OpenFileDialog1.FileName
        End If

    End Sub

    Private Sub buttonmerge_Click(sender As System.Object, e As System.EventArgs) Handles buttonmerge.Click
        If txtmergesource1.Text = "" Then

            MessageBox.Show("Please select PDF 1 file")
            Exit Sub
        End If


        If txtmergesource2.Text = "" Then

            MessageBox.Show("Please select PDF 2 file")
            Exit Sub
        End If


        strMergeSource1 = txtmergesource1.Text
        strMergeSource2 = txtmergesource2.Text
        iType = 0
        DialogResult = Windows.Forms.DialogResult.OK

        Close()
    End Sub

    Private Sub MergeSplitDeletePDF_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load

    End Sub

    Private Sub button3_Click(sender As System.Object, e As System.EventArgs) Handles button3.Click
        OpenFileDialog1.Filter = "PDF File|*.PDF||"
        If OpenFileDialog1.ShowDialog() = System.Windows.Forms.DialogResult.OK Then

            txtsplitsource.Text = OpenFileDialog1.FileName
        End If
    End Sub

    Private Sub button4_Click(sender As System.Object, e As System.EventArgs) Handles button4.Click
        If txtsplitsource.Text = "" Then

            MessageBox.Show("Please select PDF File")
            Exit Sub
        End If


        strSplitSource = txtsplitsource.Text
        iSplitStartPage = Int32.Parse(txtsplitfrompage.Text)

        iType = 1
        DialogResult = Windows.Forms.DialogResult.OK

        Close()
    End Sub

    Private Sub button5_Click(sender As System.Object, e As System.EventArgs) Handles button5.Click
        OpenFileDialog1.Filter = "PDF File|*.PDF||"
        If OpenFileDialog1.ShowDialog() = System.Windows.Forms.DialogResult.OK Then

            txtdeletepagesource.Text = OpenFileDialog1.FileName
        End If
    End Sub

    Private Sub button6_Click(sender As System.Object, e As System.EventArgs) Handles button6.Click
        If txtdeletepagesource.Text = "" Then

            MessageBox.Show("Please select PDF File")
            Exit Sub
        End If


        strDeletePageSource = txtdeletepagesource.Text
        iDeletePageNo = Int32.Parse(txtdeletepage.Text)

        iType = 2
        DialogResult = Windows.Forms.DialogResult.OK

        Close()
    End Sub
End Class

File: C#2019/Viewer/MergeSplitDeletePDF.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace PDFViewerDemo
{
    public partial class MergeSplitDeletePDF : Form
    {
        public int iType { get; set; }
        public string strMergeSource1 { get; set; }
        public string strMergeSource2 { get; set; }

        public string strSplitSource { get; set; }
        public int iSplitStartPage { get; set; }

        public string strDeletePageSource { get; set; }
        public int iDeletePageNo { get; set; }


        public MergeSplitDeletePDF()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
              openFileDialog1.Filter = "PDF File|*.PDF||";
              if (openFileDialog1.ShowDialog() == System.Windows.Forms.DialogResult.OK)
              {
                  txtmergesource1.Text = openFileDialog1.FileName;
              }
         
        }

        private void button2_Click(object sender, EventArgs e)
        {
            openFileDialog1.Filter = "PDF File|*.PDF||";
            if (openFileDialog1.ShowDialog() == System.Windows.Forms.DialogResult.OK)
            {
                txtmergesource2.Text = openFileDialog1.FileName;
            }
        }

        private void button3_Click(object sender, EventArgs e)
        {
            openFileDialog1.Filter = "PDF File|*.PDF||";
            if (openFileDialog1.ShowDialog() == System.Windows.Forms.DialogResult.OK)
            {
              txtsplitsource.Text = openFileDialog1.FileName;
            }
         
        }

        private void buttonmerge_Click(object sender, EventArgs e)
        {
            if (txtmergesource1.Text == "")
            {
                MessageBox.Show("Please select PDF 1 file");
                return;
            }

            if (txtmergesource2.Text == "")
            {
                MessageBox.Show("Please select PDF 2 file");
                return;
            }

            strMergeSource1 = txtmergesource1.Text;
            strMergeSource2 = txtmergesource2.Text;
            iType = 0;
            DialogResult = System.Windows.Forms.DialogResult.OK;
            this.Close();
        }

        private void button4_Click(object sender, EventArgs e)
        {
            if (txtsplitsource.Text == "")
            {
                MessageBox.Show("Please select PDF File");
                return;
            }

            strSplitSource = this.txtsplitsource.Text;
            iSplitStartPage = Int32.Parse(this.txtsplitfrompage.Text);
            DialogResult = System.Windows.Forms.DialogResult.OK;
            iType = 1;
            this.Close();
        }

        private void button5_Click(object sender, EventArgs e)
        {
            openFileDialog1.Filter = "PDF File|*.PDF||";
            if (openFileDialog1.ShowDialog() == System.Windows.Forms.DialogResult.OK)
            {
                txtdeletepagesource.Text = openFileDialog1.FileName;
            }
        }

        private void button6_Click(object sender, EventArgs e)
        {
            if (txtdeletepagesource.Text == "")
            {
                MessageBox.Show("Please select PDF File");
                return;
            }

            strDeletePageSource = txtdeletepagesource.Text;
            iDeletePageNo = Int32.Parse(this.txtdeletepage.Text);
            DialogResult = System.Windows.Forms.DialogResult.OK;
            iType = 2;
            this.Close();
        }
    }
}
